### PR TITLE
[SandboxIR] Implement ConstantFP

### PR DESCRIPTION
**Pull request by vporpo**
This patch implements sandboxir::ConstantFP mirroring llvm::ConstantFP.

### CODE REVIEW

File: llvm/include/llvm/SandboxIR/SandboxIR.h [hunk 1]
```diff
@@ -597,6 +598,92 @@ class ConstantInt : public Constant {
 #endif
 };
 
+class ConstantFP final : public Constant {
```
Comment (tmsri) [inline llvm/include/llvm/SandboxIR/SandboxIR.h:601]: Why are we not adding ConstantData?  If we are shadowing LLVM IR why not do it rigorously?

File: llvm/include/llvm/SandboxIR/SandboxIR.h [hunk 2]
```diff
@@ -597,6 +598,92 @@ class ConstantInt : public Constant {
...
+  bool isExactlyValue(double V) const {
+    return cast<llvm::ConstantFP>(Val)->isExactlyValue(V);
+  }
+
+  /// For isa/dyn_cast.
+  static bool classof(const sandboxir::Value *From) {
+    return From->getSubclassID() == ClassID::ConstantFP;
+  }
+
+  unsigned getUseOperandNo(const Use &Use) const final {
```
Comment (tmsri) [inline llvm/include/llvm/SandboxIR/SandboxIR.h:673]: I am trying to match this with LLVM IR.  Why not just call this "getOperandNo" with the Use as parameter?

File: llvm/include/llvm/SandboxIR/SandboxIR.h (hunk 1 above)
Comment (vporpo) [reply llvm/include/llvm/SandboxIR/SandboxIR.h:601]: We can add it in, I will add a TODO.

File: llvm/include/llvm/SandboxIR/SandboxIR.h (hunk 2 above)
Comment (vporpo) [reply llvm/include/llvm/SandboxIR/SandboxIR.h:673]: This is a SandboxIR-specific function, it is not wrapping any LLVM IR one. But yeah `getOperandNo(const Use &)` is a much better name.

**Review by tmsri — APPROVED** [review]
LGTM otherwise, having TODOs in all places where you expect a cleanup would be very useful.

**Pull request merged**